mount/losetup_linux.go

```diff
@@ -78,12 +103,50 @@ func setupLoopDev(backingFile, loopDev string, param LoopParams) (_ *os.File, re
 	if err != nil {
 		return nil, fmt.Errorf("could not open loop device: %s: %w", loopDev, err)
 	}
+
+	// LOOP_CONFIGURE is not available on Linux versions prior to 5.8, test
+	// once to see if it works
+	configOnce.Do(func() {
+		config := LoopConfig{
+			Fd: math.MaxUint32,
+		}
+
+		if err := IoctlConfigure(int(loop.Fd()), &config); err == syscall.EBADF {
+			useConfigure = true
+		}
+	})
+
 	defer func() {
 		if retErr != nil {
 			loop.Close()
 		}
 	}()
 
+	if useConfigure {
+		config := LoopConfig{
+			Fd: uint32(back.Fd()),
+		}
+
+		copy(config.Info.File_name[:], backingFile)
+		if param.Readonly {
+			config.Info.Flags |= unix.LO_FLAGS_READ_ONLY
+		}
+
+		if param.Autoclear {
+			config.Info.Flags |= unix.LO_FLAGS_AUTOCLEAR
+		}
+
+		if param.Direct {
+			config.Info.Flags |= unix.LO_FLAGS_DIRECT_IO
+		}
+
+		if err := IoctlConfigure(int(loop.Fd()), &config); err != nil {
+			return nil, fmt.Errorf("failed to configure loop device: %s: %w", loopDev, err)
```

Comment:
I checked the code in kernel, it associates the file with the loop device after all checks have succeeded. There is no point where it can fail after that.